File: C, C++/알고리즘/SelectionSort.cpp
#include<stdio.h>


int main() {

	int arr[] = { 8,4,6,5,2,9,7,3,1 };

	//선택 정렬
	int index = 0;
	int min = 0;
	for (int i = 0; i < sizeof(arr) / sizeof(int); i++) {
		
		min = arr[i];
		index = i;

		for (int j = i+1; j < sizeof(arr) / sizeof(int); j++) {		

			if (min >  arr[j]){
				min = arr[j];
				index = j;
			}		
		}
		// swap
		arr[index] = arr[i];
		arr[i] = min;
	}


	for (int i = 0; i < sizeof(arr) / sizeof(int) ; i++) {
		printf("%d  ", arr[i]);
	}


}

File: Java/javaPractice/javaPrac/src/Sorts/SelectionSort.java
package Sorts;

import java.util.Arrays;

public class SelectionSort {

	public static void main(String[] args) {
		int arr[] = {5,9,8,3,2,1,7,6,4};
		System.out.println(Arrays.toString(arr));
		 selectionSort(arr);
		System.out.println(Arrays.toString(arr));

	}

	static void selectionSort(int[] arr) {
		selectionSort(arr, 0);
	}
	
	static void selectionSort(int[] arr, int start) {
		
		if(start<arr.length-1) {
			int index = start;
			
			for(int i = start; i<arr.length; i++) {					
				if( arr[index] > arr[i] ) {
					index = i;
				}				
			}
			swap(arr,start,index);
			
			selectionSort(arr, start+1);
		}
	}
	
	public static void swap(int[] arr, int index1, int index2) {
		int num = arr[index1];
		arr[index1] = arr[index2];
		arr[index2] = num;
	}
}
				